### ISSUE TITLE
Understanding about RL on Object Detection

### ISSUE THREAD

**Issue by LaFeuilleMorte** [om-ai-lab/VLM-R1#82]
Thanks for this impressive work. I already did the GRPO training on the Qwen2.5-3B-Instruct base model. Here's some opinions on this area:

- First, the IOU reward, from my perspective, has little difference from directly using IOU loss. 
- Second, the format reward, maybe not helping at all, because Qwen2.5-3B-Instruct could output the right answer even without longer CoT, Object detection was quite easy for vision models and would rarely need "Reasoning".
- Third, The **KEY REASON** that makes so much difference between RL and SFT, I think, is the KL divergence between the Ref policy model and the policy model (Aka the VLM model). The KL divergence will constraints the VLM model not to update so aggressively from the Base model, So as the training steps goes, the RL model won't collapse on the RefGTA images. 
- Fourth, the baseline of Qwen2.5-3B-Instruct on both validation datasets was already very high, so the RL training scheme helps a little, but not too much. So basicly, I think the RL training scheme was a kind of "Robust" training method, it won't destroy the original distribution of the base model. But in such an easy object detection scenario, maybe RL not helps a lot.
- Fifth, the code seemed to have some bugs, there're chances that even the model output the right answer, the regex pattern failed to capture the right answer, like here in the evaluation json:
 `{
      "question": "Please provide the bounding box coordinate of the region this sentence describes: a man with a blue jacket and a white shirt.",
      "ground_truth": [
        653.29868352,
        517.003344,
        708.3757213440001,
        674.41406976
      ],
      "model_output": "<think>\nThe given instruction is to find the bounding box coordinates for a man wearing a blue jacket and a white shirt. In the image, there is a person walking on the sidewalk who appears to be wearing a blue jacket over a white shirt. This individual is located towards the center-left of the image.\n</think>\n<answer>\n{\n    \"bbox_2d\": [\n        (653, 534, 718, 683)\n    ]\n}\n</answer>",
      "extracted_answer": [
        0,
        0,
        0,
        0
      ],
      "correct": 0
    },`

The answer was not in \<answer\> ...\</answer\> so the regex failed to capture the bbox coordinates even if the model outputs the right answer.

**Comment by LaFeuilleMorte** [comment]
I already did the ablation study about the format reward, which indicates that it has no negative impact on the model performance. So it could very likely be the KL divergence loss that makes SFT much much worse that RL

**Comment by BrightHai** [comment]
I trained the qwen2.5-vl-3B with SFT first , then VLM-R1 GRPO training for 200 steps, no val acc changed , image understanding task

**Comment by hbhdfjy** [comment]
> I already did the ablation study about the format reward, which indicates that it has no negative impact on the model performance. So it could very likely be the KL divergence loss that makes SFT much much worse that RL我已经完成了关于格式奖励的消融研究，这表明它对模型性能没有负面影响。因此，很可能正是 KL 散度损失使得 SFT 比 RL 差得多。

如果去掉格式奖励的话和加上格式奖励的准确度没有什么区别是吗，方便看一下训练时的准确率吗

**Comment by sleepyshep** [comment]
> I already did the ablation study about the format reward, which indicates that it has no negative impact on the model performance. So it could very likely be the KL divergence loss that makes SFT much much worse that RL

Hello! I also conducted ablation study on the format reward. The results show that removing the format reward negatively impacts the model's performance. Even when I used more regular expressions to match bounding boxes, the model's performance after training for 500 steps was significantly worse compared to the model that utilized the format reward.
